src/control/cmd/dmg/network.go

```diff
@@ -23,7 +23,51 @@
 
 package main
 
-// TODO: provide implementation to this placeholder
-
 // NetCmd is the struct representing the top-level network subcommand.
-type NetCmd struct{}
+type NetCmd struct {
+	Scan networkScanCmd `command:"scan" description:"Scan for network interface devices on remote servers"`
+	List networkListCmd `command:"list" description:"List all known OFI providers on remote servers that are understood by 'scan'"`
+}
+
+// networkScanCmd is the struct representing the command to scan the machine for network interface devices
+// that match the given fabric provider.
+type networkScanCmd struct {
+	logCmd
+	connectedCmd
+	FabricProvider string `short:"p" long:"provider" description:"Filter device list to those that support the given OFI provider (default is the provider specified in daos_server.yml)"`
```

Comment:
I need to adjust the string, because the daos_shell will not be using the provider in the daos_server.yml.

Reply:
This new feature will not require the admin to put any network configuration in the client configuration, right?

Reply:
That is correct.  It depends on nothing in the client config.